drivers/net/ethernet/renesas/rswitch_xenfront.c

```diff
@@ -236,8 +240,25 @@ static int rswitch_vmq_front_connect(struct net_device *dev)
 
 	err = rswitch_vmq_front_ndev_register(rdev, type, index, tx_chain_id,
 					      rx_chain_id, mac_str ? mac : NULL);
+
+	rdev->vmq_info = (void *)__get_free_page(GFP_KERNEL | __GFP_ZERO);
+	/* TODO: correct process error */
+	if (!rdev->vmq_info) {
+		pr_err("rdev->rswitch_vmq_status ERROR\n");
```

Comment:
`kfree(type)`, `kfree(mac_str)` and reverse call for `rswitch_vmq_front_ndev_register`?